platform/android/java/lib/src/org/godotengine/godot/Godot.java

```diff
@@ -1001,6 +1003,26 @@ public boolean requestPermission(String p_name) {
 		return true;
 	}
 
+	public void requestPermissions() {
+		if (Build.VERSION.SDK_INT < Build.VERSION_CODES.M) {
+			return;
+		}
+
+		PackageInfo packageInfo;
+		try {
+			packageInfo = getPackageManager().getPackageInfo(getPackageName(),
+					PackageManager.GET_PERMISSIONS);
+		} catch (PackageManager.NameNotFoundException e) {
+			e.printStackTrace();
+			return;
+		}
+
+		String[] manifestPermissions = packageInfo.requestedPermissions;
+		if (manifestPermissions == null || manifestPermissions.length == 0)
+			return;
+		requestPermissions(manifestPermissions, REQUEST_ALL_PERMISSION_REQ_CODE);
+	}
+
```

Comment:
@cagdasc Can you move the permissions request logic (including the `requestPermission(String p_name)` above) to a separate class. This would help reduce the line count in the `Godot.java` class and help simplify and clarify the logic.
You could move the code in `PermissionsUtil` class in the `utils` package.

Reply:
I have move both to `PermissionsUtil` you said.